date-time/src/commonMain/kotlin/com/splendo/kaluga/datetime/timer/TimerProvider.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+package com.splendo.kaluga.datetime.timer
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.MainScope
+import kotlin.time.Duration
+
+/** Provides timer instances. */
+object TimerProvider {
+    /**
+     * @param duration timer duration
+     * @param interval timer tick interval
+     * @param coroutineScope a parent coroutine scope for the timer
+     * @return a timer based on the system clock
+     */
+    fun monotonic(
```

Comment:
It's a bit not obvious how it works without checking comments about. Especially name.
Here is suggestion to discuss. It can be named `intervalTimer` or `recurringTimer` (so you know it ticks on interval)

Also in some cases I need to run it forever. Maybe `duration` can be replaced with repeat type?
```
sealed class TimerRepeat {
    object Forever : TimerRepeat()
    data class Until(val duration: Duration): TimerRepeat()
}
```

Reply:
`Duration.INDEFINITE` would take care of such scenarios.
renaming the timer to recurring as it's much better indeed